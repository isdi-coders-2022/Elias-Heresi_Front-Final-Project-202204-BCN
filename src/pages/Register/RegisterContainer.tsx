import styled from "styled-components";

export const RegisterContainer = styled.div`
  height: auto;
  min-height: 100vh;
  background-color: rgb(111, 219, 135);
  text-align: center;
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  div {
    display: flex;
    flex-direction: column;
    align-items: center;

    h1 {
      padding: 15px 0;
      font-weight: bold;
    }
    h2 {
      font-size: 24px;
    }

    div {
      margin-bottom: 20px;
    }

    .register__form {
      margin-bottom: 20px;
    }
  }
  img {
    width: 400px;
    height: 300px;
    object-fit: cover;
    border: 2px solid grey;
    border-radius: 20%;
  }
  @media (max-width: 800px) {
    .form__image {
      display: none;
    }
  }
`;
