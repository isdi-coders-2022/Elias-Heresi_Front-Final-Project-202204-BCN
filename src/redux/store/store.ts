import { configureStore } from "@reduxjs/toolkit";
import userReducer from "../features/userSlice";
import uiReducer from "../features/uiSlice";
import diaryReducer from "../features/diarySlice";

const store = configureStore({
  reducer: {
    user: userReducer,
    ui: uiReducer,
    diary: diaryReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
