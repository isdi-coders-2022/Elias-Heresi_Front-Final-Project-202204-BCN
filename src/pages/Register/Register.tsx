import RegisterForm from "../../components/RegisterForm/RegisterForm";
import { RegisterContainer } from "./RegisterContainer";

const Register = () => {
  return (
    <RegisterContainer>
      <img
        src="https://www.happiness.com/community/uploads/monthly_2020_08/happy-children-kids-joyful.jpg.307017a030aa7b740d260c45d39745b2.jpg"
        alt="Placeholder"
        className="register__image"
      />
      <div className="register__form">
        <h1>Bonanza</h1>
        <h2>Sign up to improve your life, one day at a time</h2>
        <RegisterForm />
      </div>
    </RegisterContainer>
  );
};

export default Register;
