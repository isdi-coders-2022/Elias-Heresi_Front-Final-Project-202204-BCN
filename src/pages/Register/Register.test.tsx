import { Provider } from "react-redux";
import store from "../../redux/store/store";
import Register from "./Register";
import { BrowserRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";

let mockLogged: boolean = false;

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: () => ({ logged: mockLogged }),
}));

const mockUseNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockUseNavigate,
}));

describe("Given the Register page component", () => {
  describe("When it's invoked and the user isn't logged in", () => {
    test("Then 2 headers and an image will be rendered", () => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <Register />
          </Provider>
        </BrowserRouter>
      );

      const searchedHeaders = screen.getAllByRole("heading");
      const searchedImages = screen.getAllByRole("img");

      const expectedHeaders = 2;
      const expectedImages = 1;

      expect(searchedHeaders).toHaveLength(expectedHeaders);
      expect(searchedImages).toHaveLength(expectedImages);
    });
  });

  describe("When invoked and the user is already logged in", () => {
    test("Then the user will be redirected to another page", () => {
      mockLogged = true;
      render(
        <BrowserRouter>
          <Provider store={store}>
            <Register />
          </Provider>
        </BrowserRouter>
      );

      expect(mockUseNavigate).toHaveBeenCalled();
    });
  });
});
