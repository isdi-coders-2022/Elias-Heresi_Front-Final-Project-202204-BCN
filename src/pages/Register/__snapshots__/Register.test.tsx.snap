// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Register page component When it's invoked Then it should always match this snapshot 1`] = `
<div
  className="sc-gsnTZi iYDljS"
>
  <img
    alt="Placeholder"
    className="register__image"
    src="https://www.happiness.com/community/uploads/monthly_2020_08/happy-children-kids-joyful.jpg.307017a030aa7b740d260c45d39745b2.jpg"
  />
  <div
    className="register__form"
  >
    <h1>
      Bonanza
    </h1>
    <h2>
      Sign up to improve your life, one day at a time
    </h2>
    <div
      className="sc-bczRLJ brPyMg"
    >
      <form
        autoComplete="off"
        className=""
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="mb-3"
        >
          <label
            className="form-label"
            htmlFor="name"
          >
            Name
          </label>
          <input
            className="form-control"
            id="name"
            onChange={[Function]}
            placeholder="Enter first name"
            type="text"
            value=""
          />
        </div>
        <div
          className="mb-3"
        >
          <label
            className="form-label"
            htmlFor="surname"
          >
            Surname
          </label>
          <input
            className="form-control"
            id="surname"
            onChange={[Function]}
            placeholder="Enter last name"
            type="text"
            value=""
          />
        </div>
        <div
          className="mb-3"
        >
          <label
            className="form-label"
            htmlFor="email"
          >
            Email address
          </label>
          <input
            className="form-control"
            id="email"
            onChange={[Function]}
            placeholder="Enter email"
            type="email"
            value=""
          />
        </div>
        <div
          className="mb-3"
        >
          <label
            className="form-label"
            htmlFor="username"
          >
            Username
          </label>
          <input
            className="form-control"
            id="username"
            onChange={[Function]}
            placeholder="Choose your username"
            type="text"
            value=""
          />
        </div>
        <div
          className="mb-3"
        >
          <label
            className="form-label"
            htmlFor="password"
          >
            Password
          </label>
          <input
            className="form-control"
            id="password"
            onChange={[Function]}
            placeholder="Choose your password"
            type="password"
            value=""
          />
        </div>
        <section
          className="container text-center"
        >
          <button
            className="btn btn-primary"
            disabled={true}
            type="submit"
          >
            Submit
          </button>
          <p>
            Already a member?
          </p>
          <button
            className="btn btn-secondary"
            disabled={false}
            type="button"
          >
            Login
          </button>
        </section>
      </form>
    </div>
  </div>
</div>
`;
